### PR TITLE
Make transitionToRoute() and intermediateTransitionTo() work in engines

### LINKED ISSUE #186: [0.2] Route.intermediateTransitionTo() doesn't prefix the routes
It needs to have the same prefixing done as the `transitionTo`.

### PR DESCRIPTION
**Pull request by dnachev**
The PR fixes #186 and #159 in v0.2 branch.

I've added tests, but the one for `intermediateTransitionTo()` is somewhat clunky. I couldn't think of a better way.

### CODE REVIEW

**Comment by trentmwillis** [comment]
@dgeb / @rwjblue thoughts on merging this and doing one final `v0.2` release? We'll likely want to port these tests forward as well (this behavior exists upstream now).

**Comment by dgeb** [comment]
@trentmwillis I'm in favor. Do you have time time to review that this PR matches the upstream changes?

File: addon/-private/utils.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+*/
+export function prefixRouteNameArg(funcName, ...args) {
+  let routeName = args[0];
+  let owner = getOwner(this);
+  let prefix = owner.mountPoint;
+
+  // only alter the routeName if it's actually referencing a route.
+  if (owner.routable && typeof routeName === 'string') {
+    if (resemblesURL(routeName)) {
+      throw new EmberError(`${funcName} cannot be used for URLs. Please use the route name instead.`);
```
Comment (trentmwillis) [inline addon/-private/utils.js]: Forgot to make this dynamic upstream...will have to do that soon.

**Comment by trentmwillis** [comment]
@dgeb here's the upstream PR for reference: https://github.com/emberjs/ember.js/pull/14262.

A quick look over shows that `Controller#replaceRoute` is missing in this PR, but that is private API so I'm unsure if we need to support it here. Otherwise, it looks solid to me.

**Comment by dgeb** [comment]
@trentmwillis Thanks for the review. And good catch about needing to make `prefixRouteNameArg` dynamic upstream.

@dnachev Could you please update `Controller#replaceRoute` to match `Controller#transitionToRoute` (just like upstream)?

**Comment by dgeb** [comment]
Merged via #207. Release in v0.2.12. Thanks @dnachev!

**Pull request closed without merging**